gdnative-core/src/core_types/mod.rs

```diff
@@ -2,6 +2,9 @@
 
 pub mod geom;
 
+// This should be a compile time option between f32 and f64
+pub(crate) type Real = f32;
```

Comment:
Unless I'm missing something, Godot at the moment only supports floats (as of [this proposal](https://github.com/godotengine/godot-proposals/issues/892)). Thus, code would be clearer to use `f32` everywhere.

Reply:
I think it's better to use `Real` as it will have to be used eventually anyways. Converting it afterwards will likely take some effort that can be spared now.

Reply:
Since it's not clear if and when this happens, I don't think we should sacrifice clarity now for a potential change in the future.

`Real` is one extra indirection to look up for users of godot-rust, and raises the question why it exists at all. I personally prefer simpler APIs, we can always change it later. You could retain the original version as its own separate commit, so we could cherry-pick that one in the future if needed.